engine/chain/internal/query/query_service.go

```diff
@@ -0,0 +1,106 @@
+package query
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/models"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/go-resty/resty/v2"
+	"github.com/rs/zerolog"
+)
+
+// Service for querying transactions.
+type Service struct {
+	logger       zerolog.Logger
+	httpClient   *resty.Client
+	url          string
+	token        string
+	deploymentID string
+}
+
+// NewQueryService creates a new query service.
+func NewQueryService(logger zerolog.Logger, httpClient *resty.Client, url, token, deploymentID string) *Service {
+	return &Service{
+		logger:       logger,
+		httpClient:   httpClient,
+		url:          url,
+		token:        token,
+		deploymentID: deploymentID,
+	}
+}
+
+// Query a transaction.
+func (s *Service) Query(ctx context.Context, txID string) (*chainmodels.TXInfo, error) {
+	if !s.validateTX(txID) {
+		return nil, spverrors.ErrInvalidTransactionID
+	}
+	req := s.httpClient.R().
+		SetContext(ctx).
+		SetHeader("Content-Type", "application/json").
+		SetResult(&chainmodels.TXInfo{}).
```

Comment:
What do you think about creating the TXInfo{} variable above and in line 68 refer to that variable instead of casting the result?

Reply:
In documentation they mention both ways. 
https://pkg.go.dev/github.com/go-resty/resty/v2#Request.SetResult

I think we should be consistent how we do this so I think we should decide and stick to this. 

So... I don't know 
- Reffering to a pointer to initially defined empty struct is more straight forward
- The `Result` seems to be more readable
- But the casting is ugly
- But in more sophisticated scenario when we set different models under some conditions, The `Result()` would be more accurate.
```golang
req := s.httpClient.R().
if someCondition {
    req.SetResponse(&SomeModel{})
}else {
    req.SetResponse(&OtherModel{})
}

response, err := req.Get("url")
switch res := response.Result().(type) {
    case *SomeModel:
         return handleSomeModelRes(res)
   case *OtherModel:
         return  handleOtherModelRes(res)
}

```

Reply:
OK :D 
Let's stick to the simpler method with "pointer referral".

Maybe in the future we'll revisit this consideration 😄